cmd/charmd/main.go

```diff
@@ -94,7 +97,37 @@ func serve(confPath string) error {
 	}
 
 	ring := bakery.NewPublicKeyRing()
-	ring.AddPublicKeyForLocation(cfg.IdentityLocation, false, conf.IdentityPublicKey)
+	if conf.IdentityPublicKey != nil {
```

Comment:
This and the part below look almost identical, perhaps they could be made into a function.